src/metkit/mars/TypeEnum.cc

```diff
@@ -18,104 +18,137 @@
 #include "metkit/mars/TypesFactory.h"
 
 
-namespace metkit {
-namespace mars {
+namespace metkit::mars {
 
 //----------------------------------------------------------------------------------------------------------------------
 
-TypeEnum::TypeEnum(const std::string& name, const eckit::Value& settings) : Type(name, settings) {
-
-    LOG_DEBUG_LIB(LibMetkit) << "TypeEnum name=" << name << " settings=" << settings << std::endl;
-
-    eckit::Value values = settings["values"];
-    bool uppercase      = false;
-    if (settings.contains("uppercase")) {
-        uppercase = settings["uppercase"];
+void TypeEnum::addValue(const std::string& vv, uint16_t idx, bool allowDuplicates) const {
```

Comment:
signature is `const std::string& value, uint16_t idx, bool allowDuplicates` but here becomes `const std::string& vv ...`
I'd do `std::string value, uint16_t idx, bool allowDuplicates` and `value = eckit::StringTools::lower(value)`